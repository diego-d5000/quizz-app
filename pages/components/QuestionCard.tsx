import unescape from 'lodash/unescape';

const QuestionCard = ({
  category,
  question,
  questionIndicator,
  onSelectAnswer,
}: QuestionCardProps) => {
  return (
    <article className="flex flex-col justify-evenly bg-white rounded-lg p-5 text-black text-center">
      <div className="pb-8">
        <small className="text-gray-400 font-bold mb-3">{questionIndicator}</small>
        <br />
        <small className="text-gray-600">{category}</small>
        <h1 className="leading-tight font-semibold sm:text-3xl text-xl">{unescape(question)}</h1>
      </div>
      <div className="h-full flex flex-col justify-between space-y-5">
        <div>
          <button className="btn-quiz" onClick={() => onSelectAnswer('True')}>
            True
          </button>
        </div>
        <div>
          <button className="btn-quiz" onClick={() => onSelectAnswer('False')}>
            False
          </button>
        </div>
      </div>
    </article>
  );
};

interface QuestionCardProps {
  category: string;
  question: string;
  questionIndicator: string;
  onSelectAnswer: Function;
}

export default QuestionCard;
