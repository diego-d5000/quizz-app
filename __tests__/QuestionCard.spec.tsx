import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import QuestionCard from '../components/QuestionCard';

describe('QuestionCard Component', () => {
  let TestingComponent;
  const mockOnSelectAnswerCb = jest.fn();
  const mockData = {
    category: 'Entertainment: Video Games',
    type: 'boolean',
    difficulty: 'hard',
    question: 'Unturned originally started as a Roblox game.',
    correct_answer: 'True',
    incorrect_answers: ['False'],
  };

  beforeEach(() => {
    TestingComponent = render(
      <QuestionCard
        question={mockData.question}
        questionIndicator={'1/2'}
        category={mockData.category}
        onSelectAnswer={mockOnSelectAnswerCb}
      />
    );
  });

  it('snapshot', () => {
    const tree = renderer
      .create(
        <QuestionCard
          question={mockData.question}
          questionIndicator={'1/2'}
          category={mockData.category}
          onSelectAnswer={mockOnSelectAnswerCb}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('shows question in the card', () => {
    const { getByText } = TestingComponent;
    const questionElement = getByText(mockData.question);
    expect(questionElement).toBeInTheDocument();
  });

  it('sends user answer to callback', () => {
    const { getByText } = TestingComponent;
    fireEvent.click(getByText('True'));

    expect(mockOnSelectAnswerCb).toBeCalledWith('True');
  });
});
