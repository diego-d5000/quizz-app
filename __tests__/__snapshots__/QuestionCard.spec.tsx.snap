// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionCard Component snapshot 1`] = `
<article
  className="flex flex-col justify-evenly bg-white rounded-lg p-5 text-black text-center"
>
  <div
    className="pb-8"
  >
    <small
      className="text-gray-400 font-bold mb-3"
    >
      1/2
    </small>
    <br />
    <small
      className="text-gray-600 leading-tight"
    >
      Entertainment: Video Games
    </small>
    <h1
      className="leading-tight font-semibold sm:text-3xl text-xl break-words"
    >
      Unturned originally started as a Roblox game.
    </h1>
  </div>
  <div
    className="h-full flex flex-col justify-between space-y-5"
  >
    <div>
      <button
        className="btn-quiz"
        onClick={[Function]}
      >
        True
      </button>
    </div>
    <div>
      <button
        className="btn-quiz"
        onClick={[Function]}
      >
        False
      </button>
    </div>
  </div>
</article>
`;
